### PR TITLE
Adding vectorized implementations of Log2 to Vector64/128/256/512

### PR DESCRIPTION
**Pull request by tannergooding**
This makes progress towards https://github.com/dotnet/runtime/issues/93513 and sets up the basic infrastructure for testing the new APIs.

In general the tests are following the same logic as the scalar versions and testing the same values, for simplicity.

The algorithm is the same as being used for the `TensorPrimitives` type but isn't currently shared due to that being out of band and therefore not supporting the `ISimdVector` interface since it needs to multi-target `$(NetCoreAppPrevious)`. It's likely not worth the additional `#ifdef` at this time to have `$(NetCoreAppCurrent)` calling the APIs exposed on the vector types.

### CODE REVIEW

**Comment by stephentoub** [comment]
> The algorithm is the same as being used for the TensorPrimitives type but isn't currently shared due to that being out of band

Presumably we should we be adding a net9.0 TFM to that package and specializing the implementation for net9.0 to use these Vector64/128/256/512 APIs, right?

**Comment by tannergooding** [comment]
> Presumably we should we be adding a net9.0 TFM to that package and specializing the implementation for net9.0 to use these Vector64/128/256/512 APIs, right?

We have that currently, it's what `$(NetCoreAppCurrent)` is targeting. We certainly could consume these APIs, but it's going to involve additional `ifdefs` in many of the algorithms and it's not clear if that's worth it since we need to duplicate the logic regardless currently.

No strong preference and happy to leave it as is or update it to:
```csharp
public static Vector128<T> Invoke(Vector128<T> x)
{
#if NET9_0_OR_GREATER
    if (typeof(T) == typeof(double))
    {
        return Vector128.Log2(x.AsDouble()).As<double, T>();
    }
    else
    {
        Debug.Assert(typeof(T) == typeof(float));
        return Vector128.Log2(x.AsSingle()).As<float, T>();
    }
#else
    if (typeof(T) == typeof(double))
    {
        return Log2OperatorDouble.Invoke(x.AsDouble()).As<double, T>();
    }
    else
    {
        Debug.Assert(typeof(T) == typeof(float));
        return Log2OperatorSingle.Invoke(x.AsSingle()).As<float, T>();
    }
#endif
}
```

**Comment by stephentoub** [comment]
> No strong preference and happy to leave it as is or update it to:

Thanks. I'd prefer to use the new APIs when targeting .NET 9, in order to a) help validate them (all of the TensorPrimitives tests will accrue to the Vector APIs), b) benefit from any improvements that _might_ be possible in those core APIs because they're lower in the stack, c) be able to better keep the implementations in sync between what's in TensorPrimitives for .NET 8 and what's in these Vector APIs for 9, and d) reduce the binary size a bit for the library deployed for .NET 9.  _Eventually_, when we no longer need to build for <= .NET 8, we could also remove the custom implementations.

This could also be our answer to https://github.com/dotnet/runtime/issues/96452 if that's desirable, e.g. these would only be accelerated for .NET 9.

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.netcore.cs
Comment (stephentoub) [inline src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.netcore.cs:11798]: Should we move these into the Log2Operator implementations instead? Each of its Invokes would then presumably become one-liners?

Either way is fine, but in some cases we compose one operator on top of another, so at least for those cases, doing it in the operator itself would serve double duty.

Comment (stephentoub) [reply src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.netcore.cs:11798]: I mean instead of this ifdef here, have it within Log2OperatorDouble.Invoke
https://github.com/dotnet/runtime/pull/96455/files/52d5a663edaf4b8e52fc56fb086bd80babf9a04b..4039954727505df784b86bb68f54c2d7d48ca2db#diff-9af9871a07c3bc8868a023467dd1e92b08eaee3e04fcc490126d89263678ed8eR11923

Comment (tannergooding) [reply src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.netcore.cs:11798]: I'm not sure that's "better".

That's going to:
* add an additional indirection for the .NET 9 case
* push the `ifdef` logic into (rather than around) the already complex implementation
* be dissimilar to how end users would need to compose these operations 
  * they can't access things like `LogOperatorDouble` and so have to go through `Log<double>`
* make the diff more complex when we move to an exclusively .NET 9+ world and only need to worry about dispatching to Vector64/128/256/512

I would also guess it's unnecessary for most of the APIs since we're either specializing (in which case we likely have a customized implementation anyways) or we're generic and we'd be calling the `Log2Operator<T>` version regardless.

Comment (stephentoub) [reply src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.netcore.cs:11798]: > add an additional indirection for the .NET 9 case

Just from a compilation persective, right? I'd hope they'd result in the same resulting asm.

> push the ifdef logic into (rather than around) the already complex implementation

From my perspective, that's a benefit, in that it means anyone reusing the implementation will get the best solution. If there's only one call site, it doesn't really matter; if there's ever multiple, then it matters (e.g. ExpOperator gets used in multiple places).

> be dissimilar to how end users would need to compose these operations

I'm not following. Can you elaborate?

In any event, like I said, either way is fine.

**Comment by tannergooding** [comment]
One of the failures is a JIT assert:
```
Assert failure(PID 10027 [0x0000272b], Thread: 85988 [0x14fe4]): Assertion failed '!m_inited' in 'System.Runtime.Intrinsics.VectorMath:Log2Single[System.Runtime.Intrinsics.Vector128`1[float],System.Runtime.Intrinsics.Vector128`1[int],System.Runtime.Intrinsics.Vector128`1[uint]](System.Runtime.Intrinsics.Vector128`1[float]):System.Runtime.Intrinsics.Vector128`1[float]' during 'Importation' (IL size 976; hash 0xefb066b7; Tier0)
```

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.netcore.cs
Comment (tannergooding) [reply src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.netcore.cs:11787]: > Just from a compilation persective, right? I'd hope they'd result in the same resulting asm.

It depends. There's a number of cases where the JIT will decide to emit a jump stub for trivial forwards rather than "inlining" the whole method or simply calling the actual target directly. I haven't checked in this exact case, but I've found it better to avoid those extra jumps where possible historically speaking.

> From my perspective, that's a benefit, in that it means anyone reusing the implementation will get the best solution. If there's only one call site, it doesn't really matter; if there's ever multiple, then it matters (e.g. ExpOperator gets used in multiple places).

I'm viewing `Log2OperatorSingle` as an implementation detail of `Log2Operator<T>` which itself is an implementation detail of the public `TensorPrimitives.Log2<T>`. I'd expect that most code just calls `TensorPrimitives.Log2` directly and iff we have any internal logic that really needs to reuse `Log2` (and I'm not aware of any currently), then it should use `Log2Operator<T>`. 

This also fits the longer term goal once we hit .NET 9+ where we no longer need to duplicate the actual implementation and we can simply call `Vector128.Log2()` directly and allows us to completely remove `Log2OperatorSingle/Double`. So any code using `TensorPrimitives.Log2<T>` doesn't need to be updated, where-as we'd have to update it if we did use `Log2OperatorSingle` directly.

This then also meshes with how users will end up building/composing their own, where they likewise have to use `TensorPrimitives.Log2<T>` and `Vector128.Log2<T>`

> In any event, like I said, either way is fine.

This is one I think I'll keep as is, because of the reasons above. I just think it fits more cleanly into the longer term code shape.

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.netcore.cs
Comment (stephentoub) [inline src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.netcore.cs:11851]: Is the test project set up to run tests against .NET 8, or are we only testing the .NET 9 and netstandard2.0 bits right now? Given the non-trivial code differences involved in all three versions, it'd probably be best if we ran all three sets of tests... I _think_ right now we're only running two, at least locally (maybe somehow CI runs all of them?)

Comment (tannergooding) [reply src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.netcore.cs:11851]: I'm unsure about which it runs currently. I agree it'd be beneficial if all are running, even if some are in the outer loop.

**Review by stephentoub — APPROVED** [review]

File: src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.netcore.cs
Comment (stephentoub) [reply src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.netcore.cs:11851]: @ericstj, @ViktorHofer, I understand from Viktor that this is currently not possible. We're going to have a large amount of code in this library that differs between its netstandard, net8, and net9 assets, and really need to run the tests on all three. How can we make that happen in some form?

Comment (ViktorHofer) [reply src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.netcore.cs:11851]: Steve and I had a conversation about this offline. Being able to test previous .NETCoreApp versions is tracked via https://github.com/dotnet/runtime/issues/54639.

> here are couple things that would need to change:
Obviously, all referenced projects / packages need to support that older .NETCoreApp TFM. I.e. RemoteExecutor currently doesn't (for no particular reason). I have a PR open that changes that: [Update Arcade TFMs according to TFM docs by ViktorHofer · Pull Request #14332 · dotnet/arcade (github.com)](https://github.com/dotnet/arcade/pull/14332/files#diff-f474402363c7ba47c119ded0c56143aa9821e404cda6fd9e95da1595f70f116c). TestUtilities.csproj in runtime needs to change as well.
Unfortunately, when testing .NETCoreApp our runtime specific test infra relies on the runtime being produced live. Changes are needed in the RunTests scripts and the msbuild tests files.
If you want to have those tests run in CI as well, changes to sendtohelix projects would be required as they have the same "Current" expectations hardcoded.

> FWIW there are currently two severe issues affecting running tests locally in runtime:
[Local running of libraries tests is broken · Issue #95762 · dotnet/runtime (github.com)](https://github.com/dotnet/runtime/issues/95762)
[`dotnet test` is not running tests for .NETFramework targets · Issue #94183 · dotnet/runtime (github.com)](https://github.com/dotnet/runtime/issues/94183)

Comment (ericstj) [reply src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.netcore.cs:11851]: A workaround might be to have a separate test project that uses the net8.0 build of this assembly src project, but it would be running on latest runtime since that's what our tests do.  The work to run on older frameworks is substantial as @ViktorHofer points out.

Comment (stephentoub) [reply src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.netcore.cs:11851]: > A workaround

Thanks, I'll try.

Comment (stephentoub) [reply src/libraries/System.Numerics.Tensors/src/System/Numerics/Tensors/netcore/TensorPrimitives.netcore.cs:11851]: https://github.com/dotnet/runtime/pull/96641

File: src/libraries/System.Collections.Specialized/tests/System.Collections.Specialized.Tests.csproj
Comment (tannergooding) [inline src/libraries/System.Collections.Specialized/tests/System.Collections.Specialized.Tests.csproj:9]: @vargaz, could I get sign-off on the workaround here. Same one as you recommended on the other issue.

Comment (vargaz) [reply src/libraries/System.Collections.Specialized/tests/System.Collections.Specialized.Tests.csproj:9]: Looks ok to me, still not sure why that test suite is affected though.

Comment (tannergooding) [reply src/libraries/System.Collections.Specialized/tests/System.Collections.Specialized.Tests.csproj:9]: I'm not sure either. But it's not reproing elsewhere and is otherwise blocking the PR.

Comment (tannergooding) [reply src/libraries/System.Collections.Specialized/tests/System.Collections.Specialized.Tests.csproj:9]: @vargaz, looks like it's getting a slightly different error now and still hitting an OOM later:
```
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(819,5): error : Failed to compile C:\helix\work\workitem\e\wasm_build\obj\wasm\for-build\System.Private.CoreLib.dll.bc -> C:\helix\work\workitem\e\wasm_build\obj\wasm\for-build\System.Private.CoreLib.dll.o [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(819,5): error : C:\helix\work\workitem\e\publish>C:\Windows\System32\chcp.com 65001 1>nul  [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(819,5): error : C:\helix\work\workitem\e\publish>setlocal [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(819,5): error : C:\helix\work\workitem\e\publish>set errorlevel=dummy  [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(819,5): error : C:\helix\work\workitem\e\publish>set errorlevel=  [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(819,5): error : C:\helix\work\workitem\e\publish>C:\helix\work\correlation\build\\wasi-sdk\\bin\clang.exe "@C:\helix\work\correlation\build\microsoft.netcore.app.runtime.wasi-wasm\runtimes\wasi-wasm\native\src\wasi-default.rsp" "@C:\helix\work\workitem\e\wasm_build\obj\wasm\for-build\wasi-compile-bc.rsp" -c -o "C:\Users\ContainerAdministrator\AppData\Local\Temp\tmp4k2wwb.tmp" "C:\helix\work\workitem\e\wasm_build\obj\wasm\for-build\System.Private.CoreLib.dll.bc"  [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(819,5): error : clang version 16.0.0 [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(819,5): error : Target: wasm32-unknown-wasi [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(819,5): error : Thread model: posix [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(819,5): error : InstalledDir: C:/helix/work/correlation/build//wasi-sdk//bin [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(819,5): error :  (in-process) [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(819,5): error :  "C:/helix/work/correlation/build/wasi-sdk/bin/clang.exe" -cc1 -triple wasm32-unknown-wasi -emit-obj -disable-free -clear-ast-before-backend -disable-llvm-verifier -discard-value-names -main-file-name System.Private.CoreLib.dll.bc -mrelocation-model static -mframe-pointer=none -ffp-contract=on -fno-rounding-math -mconstructor-aliases -target-cpu generic -fvisibility=hidden -mllvm -treat-scalable-fixed-error-as-warning -debug-info-kind=constructor -dwarf-version=4 -debugger-tuning=gdb -v -fcoverage-compilation-dir=C:/helix/work/workitem/e/publish -resource-dir C:/helix/work/correlation/build/wasi-sdk/lib/clang/16 -Oz -fdebug-compilation-dir=C:/helix/work/workitem/e/publish -ferror-limit 19 -fgnuc-version=4.2.1 -vectorize-slp -o "C:\\Users\\ContainerAdministrator\\AppData\\Local\\Temp\\tmp4k2wwb.tmp" -x ir "C:\\helix\\work\\workitem\\e\\wasm_build\\obj\\wasm\\for-build\\System.Private.CoreLib.dll.bc" [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(819,5): error : clang -cc1 version 16.0.0 based upon LLVM 16.0.0 default target wasm32-wasi [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(819,5): error : LLVM ERROR: out of memory [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(819,5): error : Allocation failed [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(819,5): error : PLEASE submit a bug report to https://github.com/llvm/llvm-project/issues/ and include the crash backtrace, preprocessed source, and associated run script. [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(819,5): error : Stack dump: [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(819,5): error : 0.	Program arguments: C:\\helix\\work\\correlation\\build\\\\wasi-sdk\\\\bin\\clang.exe @C:\\helix\\work\\correlation\\build\\microsoft.netcore.app.runtime.wasi-wasm\\runtimes\\wasi-wasm\\native\\src\\wasi-default.rsp @C:\\helix\\work\\workitem\\e\\wasm_build\\obj\\wasm\\for-build\\wasi-compile-bc.rsp -c -o C:\\Users\\ContainerAdministrator\\AppData\\Local\\Temp\\tmp4k2wwb.tmp C:\\helix\\work\\workitem\\e\\wasm_build\\obj\\wasm\\for-build\\System.Private.CoreLib.dll.bc [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(819,5): error : 1.	Code generation [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(819,5): error : 2.	Running pass 'Function Pass Manager' on module 'C:\helix\work\workitem\e\wasm_build\obj\wasm\for-build\System.Private.CoreLib.dll.bc'. [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(819,5): error : 3.	Running pass 'WebAssembly Explicit Locals' on function '@corlib_System_Buffers_IndexOfAnyAsciiSearcher_IndexOfAnyVectorized_System_Buffers_IndexOfAnyAsciiSearcher_DontNegate_System_Buffers_IndexOfAnyAsciiSearcher_Default_int16__int_System_Buffers_IndexOfAnyAsciiSearcher_AsciiState_' [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(819,5): error : Exception Code: 0xC000001D [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(819,5): error : 0x003CFB73 <unknown module> [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(819,5): error : 0x71B0C902 <unknown module> [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(819,5): error : 0x003B23A1 <unknown module> [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(819,5): error : 0x003B44C0 <unknown module> [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(819,5): error : 0x00C72CA9 <unknown module> [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(819,5): error : 0x00C72F30 <unknown module> [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(819,5): error : 0x00C72E9A <unknown module> [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(819,5): error : 0x00C72E2C <unknown module> [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(819,5): error : 0x0057829A <unknown module> [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(819,5): error : 0x0281BFA0 <unknown module> [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(819,5): error : 0x02714C1C <unknown module> [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(819,5): error : clang: error: clang frontend command failed due to signal (use -v to see invocation) [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(819,5): error : clang version 16.0.0 [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(819,5): error : Target: wasm32-unknown-wasi [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(819,5): error : Thread model: posix [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(819,5): error : InstalledDir: C:/helix/work/correlation/build//wasi-sdk//bin [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
C:\helix\work\correlation\build\wasm-shared\WasmApp.Common.targets(819,5): error : clang: note: diagnostic msg: Error generating preprocessed source(s) - no preprocessable inputs. [took 152.83s] [C:\helix\work\workitem\e\publish\ProxyProjectForAOTOnHelix.proj]
```

**Comment by vargaz** [comment]
One difference between the linked corlib with this PR is now Vector512 is not linked away, which causes a bunch more code to be AOTed. It looks like the problem is actually the methods added to AssertExtensions, those methods are not linked away for some reason, keeping Vector512 alive as well.

**Comment by tannergooding** [comment]
> It looks like the problem is actually the methods added to AssertExtensions, those methods are not linked away for some reason, keeping Vector512 alive as well.

I've relocated those to only be in `System.Runtime.Intrinsics.Tests` for the time being, which should hopefully help avoid the issue.

**Pull request merged**
